### DISCUSSION TITLE
DJL v0.11.0 release note

### DISCUSSION THREAD

**Discussion by frankfliu** [deepjavalibrary/djl#934] [News]
DJL v0.11.0 brings the new engines [XGBoost](https://github.com/deepjavalibrary/djl/tree/master/ml/xgboost) 1.3.1, updates [PyTorch](https://github.com/awslabs/djl/tree/master/pytorch/pytorch-engine) to 1.8.1, [TensorFlow](https://github.com/deepjavalibrary/djl/tree/master/tensorflow/tensorflow-engine) to 2.4.1, [Apache MXNet](https://github.com/deepjavalibrary/djl/blob/master/mxnet/mxnet-engine/README.md) 1.8.0, [PaddlePaddle](https://github.com/deepjavalibrary/djl/tree/master/paddlepaddle/paddlepaddle-engine) to 2.0.2 and introduces several new features:

## Key Features

* Supports [XGBoost](https://github.com/deepjavalibrary/djl/tree/master/ml/xgboost) 1.3.1 engine inference: now you can run prediction using models trained in XGBoost.
* Upgrades [PyTorch](https://github.com/awslabs/djl/tree/master/pytorch/pytorch-engine) to 1.8.1 with CUDA 11.1 support.
* Upgrades [TensorFlow](https://github.com/deepjavalibrary/djl/tree/master/tensorflow/tensorflow-engine) to 2.4.1 with CUDA 11.0 support.
* Upgrades [Apache MXNet](https://github.com/deepjavalibrary/djl/blob/master/mxnet/mxnet-engine/README.md) to 1.8.0 with CUDA 11.0 support.
* Upgrades [PaddlePaddle](https://github.com/deepjavalibrary/djl/tree/master/paddlepaddle/paddlepaddle-engine) to 2.0.2.
* Upgrades [SentencePiece](https://github.com/deepjavalibrary/djl/tree/master/extensions/sentencepiece) to 0.1.95.
* Introduces the djl-serving brew package: now you can install djl-serving with `brew install djl-serving`.
* Introduces the djl-serving plugins.
* Introduces [Amazon Elastic Inference](https://aws.amazon.com/machine-learning/elastic-inference/) support.

## Enhancement

* Improves TensorFlow performance by reducing GC and fixed memory leaking issue ([#892](https://github.com/awslabs/djl/pull/892))
* djl-serving now can run all the engines out-of-box ([#886](https://github.com/awslabs/djl/pull/886))
* Improves DJL training by using multi-threading on each GPU ([#743](https://github.com/awslabs/djl/pull/743))
* Implements several operators:
    * Adds boolean set method to NDArray ([#784](https://github.com/awslabs/djl/pull/784))
    * Adds batch dot product operator ([#849](https://github.com/awslabs/djl/pull/849))
    * Adds norm operator to PyTorch ([#692](https://github.com/awslabs/djl/pull/692))
    * Adds one hot operator ([#684](https://github.com/awslabs/djl/pull/684))
    * Adds weight decay to Loss ([#788](https://github.com/awslabs/djl/pull/788))
* Adds setGraphExecutorOptimize option for PyTorch engine. ([#904](https://github.com/awslabs/djl/pull/904))
* Introduces String tensor support for ONNXRuntime ([#724](https://github.com/awslabs/djl/pull/724))
* Introduces several API improvements
    * Creates ObjectDetectionDataset ([#683](https://github.com/awslabs/djl/pull/683))
    * Improves Block usability ([#712](https://github.com/awslabs/djl/pull/712))
    * Adds BlockFactory feature in model loading ([#805](https://github.com/awslabs/djl/pull/805))
    * Allows PyTorch stream model loading ([#729](https://github.com/awslabs/djl/pull/729))
    * Adds NDList decode from InputStream ([#734](https://github.com/awslabs/djl/pull/734))
    * Adds SymbolBlock Serialization ([#687](https://github.com/awslabs/djl/pull/687))
* Introduces model searching feature in djl central ([#799](https://github.com/awslabs/djl/pull/799))

## Documentation and examples

* Introduces DJL tutorials - [How to load model](https://youtu.be/POIVT0L_P58) on DJL Youtube Channel
* Adds the PaddlePaddle load model documentation ([#811](https://github.com/awslabs/djl/pull/811))
* Adds the documentations for profiler ([#722](https://github.com/awslabs/djl/pull/722))
* Adds face detection and face recognition examples ([#814](https://github.com/awslabs/djl/pull/814))
* Adds [model training visualization demo](https://github.com/deepjavalibrary/djl-demo/tree/master/visualization-vue) using Vue

## Breaking change

* Renames CheckpointsTrainingListener to SaveModelTrainingListener ([#686](https://github.com/awslabs/djl/pull/686))
* Removes erroneous random forest application ([#726](https://github.com/awslabs/djl/pull/726))
* Deletes DataManager class ([#691](https://github.com/awslabs/djl/pull/691))
* Classes under ai.djl.basicdataset packages has been moved into each sub-packages.

## Bug Fixes

* Fixes BufferOverflowException when handling handling subimage ([#866](https://github.com/awslabs/djl/pull/866))
* Fixes ONNXRuntime 2nd engine dependency from IrisTranslator ([#853](https://github.com/awslabs/djl/pull/853))
* Fixes sequenceMask error when n dimension is 2 ([#828](https://github.com/awslabs/djl/pull/828))
* Fixes TCP port range buf in djl-serving ([#773](https://github.com/awslabs/djl/pull/773))
* Fixes one array case for concat operator ([#739](https://github.com/awslabs/djl/pull/739))
* Fixes non-zero operator for PyTorch ([#704](https://github.com/awslabs/djl/pull/704))

## Known issues

* TensorFlow engine has known memory leak issue due to JavaCPP dependency. The memory leak issue has been fixed in javacpp 1.5.6-SNAPSHOT. User has to manually include javacpp 1.5.6-SNAPSHOT to avoid memory leak. See: https://github.com/deepjavalibrary/djl/tree/master/tensorflow/tensorflow-engine#installation for more detail.

## Contributors

This release is thanks to the following contributors:

* Akshay Rajvanshi(@aksrajvanshi (https://github.com/ghost))
* Anthony Feenster(@anfee1 (https://github.com/anfee1))
* Calvin(@mymagicpower (https://github.com/mymagicpower))
* enpasos(@enpasos (https://github.com/enpasos))
* Erik Bamberg(@ebamberg (https://github.com/ebamberg))
* Frank Liu(@frankfliu (https://github.com/frankfliu))
* G Goswami(@goswamig (https://github.com/goswamig))
* Hugo Miguel Ferreira(@hmf (https://github.com/hmf))
* Hodovo(@Hodovo (https://github.com/Hodovo))
* Jake Lee(@stu1130 (https://github.com/stu1130))
* Lai Wei(@roywei (https://github.com/roywei))
* Qing Lan(@lanking520 (https://github.com/lanking520))
* Marcos(@markbookk (https://github.com/markbookk))
* Stan Kirdey(@skirdey (https://github.com/skirdey))
* Zach Kimberg(@zachgk (https://github.com/zachgk))
* 付颖志 (@fuyz (https://github.com/fuyz))
* 石晓伟(@Shixiaowei02 (https://github.com/Shixiaowei02))

<hr /><em>This discussion was created from the release <a href='https://github.com/deepjavalibrary/djl/releases/tag/v0.11.0'>DJL v0.11.0 release note</a>.</em>
